### PR TITLE
Refactor test workflow

### PR DESCRIPTION
**Pull request by eyurtsev**
# Refactor the test workflow

This PR refactors the tests to run using a single test workflow. This makes it easier to relaunch failing tests and see in the UI which test failed since the jobs are grouped together.

## Who can review?

### CODE REVIEW

**Comment by dev2049** [comment]
will they still be run in parallel?

**Comment by eyurtsev** [comment]
@dev2049 yeah all envs run in parallel when using a matrix strategy

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,15 +18,27 @@ jobs:
...
+                poetry install
+              else
+                echo "Running extended tests, installing dependencies with poetry..."
+                poetry install -E extended_testing
+              fi
+      - name: Run $${{matrix.test_type}} tests
         run: |
-          make test
+          make tests
+        shell: bash
```
Comment (hwchase17) [inline .github/workflows/test.yml:44]: why is shell: bash needed?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -18,15 +18,27 @@ jobs:
...
+                echo "Running core tests, installing dependencies with poetry..."
+                poetry install
+              else
+                echo "Running extended tests, installing dependencies with poetry..."
+                poetry install -E extended_testing
+              fi
+      - name: Run $${{matrix.test_type}} tests
         run: |
-          make test
+          make tests
```
Comment (dev2049) [inline .github/workflows/test.yml:43]: why the rename, and why do we have both?

Comment (dev2049) [reply .github/workflows/test.yml:43]: as in why rename here, and why do we have both in makefile

**Review by dev2049 — APPROVED** [review]
one qq, otherwise lgtm!

File: .github/workflows/test.yml (hunk 1 above)
Comment (eyurtsev) [reply .github/workflows/test.yml:44]: Rule of thumb is to always specify the shell in all workflows.  The shell turns on strict mode which can make it much faster to debug.

File: .github/workflows/test.yml (hunk 2 above)
Comment (eyurtsev) [reply .github/workflows/test.yml:43]: Oh rename is totally accidental, but it does the same thing. I'll follow up with a minor typo fix in a second after landing this.

**Pull request merged**